src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1027,12 +1044,28 @@ private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final Strin
      */
     private Map<String, Map<String, String>> loadAliases(final ResourceResolver resolver) {
         final Map<String, Map<String, String>> map = new ConcurrentHashMap<>();
-        final String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
-        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
-        while (i.hasNext()) {
-            final Resource resource = i.next();
-            loadAlias(resource, map);
-        }
+        String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
+		if (this.factory.isForceNoAliasTraversal()) {
+			queryString += " option(traversal fail)";
+		}
+		while (true){
+	        try {
+		        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
+		        while (i.hasNext()) {
+		            final Resource resource = i.next();
+		            loadAlias(resource, map);
+		        }
+		        break;
+	        } catch (Exception e) {
```

Comment:
Is `Exception` the right thing to catch here? I think we can use more fine-grained exceptions.  Also what happens to resolution calls when initialisation never completes?

Reply:
@rombert  - I did try to find an appropriate answer and honestly did some cargo culting on that one as the cases handling failing traversals that I could find did do it that way - I'll crosscheck the oak codebase if we have a reliable behavior - esp covering the case as well where the queryengine doesn't understand the option for more refined logmessage.